src/Nether.Analytics/OutputFormatters/CsvOutputFormatter.cs

```diff
@@ -60,11 +60,20 @@ public string Format(Message message)
             return builder.ToString();
         }
 
+        public string FormatWithHeaders(Message msg)
+        {
+            var sb = new StringBuilder();
+            sb.AppendLine(Header);
+            sb.Append(Format(msg));
+
+            return sb.ToString();
+        }
+
         public string Header
         {
             get
             {
-                return string.Join(Separator.ToString(), Columns) + Environment.NewLine;
```

Comment:
Agreed. I'll also attempt to remove the new line in the Payload (and fix all the output providers accordingly).